lib/graph_utils.py

```diff
@@ -12,17 +13,29 @@ def visit_node(node, depth = 0):
         '''Recursively crawls AST and adds relevant nodes'''
         if not header:
             name = node.spelling
-            
+
             if depth == 0 or (filename.name in str(node.location.file)):
                 symbols.add(name)
                 for child in node.get_children():
                     visit_node(child, 1)
 
         else:
             name = node.spelling
-            symbols.add(name)
+
+            if node.kind in {
+                CursorKind.STRUCT_DECL, CursorKind.ENUM_DECL, CursorKind.TYPEDEF_DECL,
+                CursorKind.CLASS_DECL, CursorKind.FUNCTION_DECL,
+                CursorKind.MACRO_DEFINITION, CursorKind.FIELD_DECL,
+            }:
```

Comment:
It might be good to create this set as a global frozenset. The name would be a bit more descriptive.

Reply:
also, this hunk of the commit seems based on #2